monkey/infection_monkey/monkey.py

```diff
@@ -477,9 +477,9 @@ def _close_tunnel(self):
 
     def _send_log(self):
         logger.info("Sending agent logs to the Island")
-        monkey_log_path = get_agent_log_path()
-        if monkey_log_path.is_file():
-            with open(monkey_log_path, "r") as f:
+        agent_log_path = create_secure_agent_log_file()
+        if agent_log_path.is_file():
+            with open(agent_log_path, "r") as f:
```

Comment:
This is obviously problematic. This works because of the caching, but obviously this code shouldn't be calling `create()`. We need to come up with a better way of structuring monkey_log_path.py to avoid this.